core/src/main/scala/org/apache/spark/ui/exec/ExecutorsPage.scala

```diff
@@ -52,12 +52,17 @@ private[ui] class ExecutorsPage(
   private val listener = parent.listener
 
   def render(request: HttpServletRequest): Seq[Node] = {
-    val storageStatusList = listener.storageStatusList
+    val (storageStatusList, execInfo) = listener.synchronized {
```

Comment:
I would add a comment to explain why we need to do this in synchronized though, and include issue number